### DISCUSSION TITLE
Nuxt "Maximum call stack size exceeded" error for utils.

### DISCUSSION THREAD

**Discussion by RajinCoding2007** [nuxt/nuxt#20475] [Questions] [answered]
I was testing how utils work in Nuxt, that's why I have created two files `utils/printPath.js` and `utils/callPrintPath.js`. `utils/printPath.js` is calling function of `utils/callPrintPath.js` and `utils/callPrintPath.js` is calling function of `utils/printPath.js`. They are calling each other. So, this error occurred `Maximum call stack size exceeded`. So, how does `utils` work in Nuxt? Which `utils` function is imported before? How can both utils have access of each other?

**Comment by ndragun92** [comment]
It would be super helpful if you make a reproduction in StackBlitz or somewhere else or share those parts of the code. Feels like u are calling an function that runs into loop and causes memory leak issues

**Comment by RajinCoding2007** [comment]
I am also saying this. It causes an infinite loop.

**Reply by warflash** [reply, marked as answer]
Like @ndragun92 said, it's a lot easier to figure out what the issue is with a reproduction at hand. Based on what you're describing you have a circular dependency in your function so call stack exceeds would be expected imo. Perhaps you forgot a return statement somewhere in case you intentionally are doing recursive calls
